### ISSUE TITLE
Multiple/language dictionaries

### ISSUE THREAD

**Issue by ketralnis** [atom/spell-check#11]
Need the ability to specify dictionary language. Most OSs these days ship with their own and even know what the user's preference is already.

**Comment by celsomiranda** [comment]
+1
Spellchecking is badly broken on Windows.
Maybe we could use a system like Sublime, that enables the user to select which language to use. This would be easy to expose through the `ctrl+p` panel or in the package settings.

**Comment by paulcbetts** [comment]
In the Slack app, we do this via detecting the keyboard layout the user is typing in, no configuration necessary. I've done several PRs to keyboard-layout and node-spellchecker to make this possible, Atom just needs to combine the pieces together.

**Comment by abeldantas** [comment]
how do I turn spell-check on in a json file under windows?

**Comment by bentolor** [comment]
Stumbling over the same issue: As german developer working under Linux I mostly write a mixture of English & German. Currently there seems to be even no way to switch to German at all due to #26

**Comment by baldurbjarnason** [comment]
I'm on a relatively (i.e. few days) new Macbook Pro running Yosemite 10.10.3 and Atom doesn't respect the language I've set in my Settings. I've set it to British English but it quite obviously is using a US dictionary.

How to replicate it? Considering this is a fresh install of Atom on a freshly updated, new, laptop and the only thing I'm doing is writing in a markdown file, I'm not sure how much more I can do.

**Comment by borisovg** [comment]
+1 (Debian GNU/Linux, en_GB)

**Comment by arturhoo** [comment]
It'd be nice to have an option to select the dictionary from within atom itself. As a brazilian developer, I write Portuguese and English every day on a Mac without ever switching the keyboard or language preferences within Mac OS.

As a reference, on Sublime Text, the dictionary selection can be done through the `View` menu.

**Comment by tommarmstrong** [comment]
+1 for British English

**Comment by bjornregnell** [comment]
+1 for multi-language spell check: I mix two languages all the time (swedish+english) - in notepad++ it is possible to have two dictionaries active at the same time, causing only words that are neither dictionary underlined as misspelled, and suggestions in correction drop-down include a merge of similar words from both languages.

**Comment by victorlcampos** [comment]
1+ Brazilian Portuguese

**Comment by Munchotaur** [comment]
+1 for British English; would like access to French and Greek too

**Comment by thomashunziker** [comment]
+1 Add other languages. Why is not the chrome feature used for it?

**Comment by nlw0** [comment]
Who knew in the atomic age we would all be forced to speak english...

**Comment by hadim** [comment]
Any news here ?

```sh
 $ hunspell -D
SEARCH PATH:
.::/usr/share/hunspell:/usr/share/myspell:/usr/share/myspell/dicts:/Library/Spelling:/home/hadim/.openoffice.org/3/user/wordbook:.openoffice.org2/user/wordbook:.openoffice.org2.0/user/wordbook:Library/Spelling:/opt/openoffice.org/basis3.0/share/dict/ooo:/usr/lib/openoffice.org/basis3.0/share/dict/ooo:/opt/openoffice.org2.4/share/dict/ooo:/usr/lib/openoffice.org2.4/share/dict/ooo:/opt/openoffice.org2.3/share/dict/ooo:/usr/lib/openoffice.org2.3/share/dict/ooo:/opt/openoffice.org2.2/share/dict/ooo:/usr/lib/openoffice.org2.2/share/dict/ooo:/opt/openoffice.org2.1/share/dict/ooo:/usr/lib/openoffice.org2.1/share/dict/ooo:/opt/openoffice.org2.0/share/dict/ooo:/usr/lib/openoffice.org2.0/share/dict/ooo
AVAILABLE DICTIONARIES (path is not mandatory for -d option):
/usr/share/hunspell/en_ZA
/usr/share/hunspell/fr_LU
/usr/share/hunspell/en_AU
/usr/share/hunspell/fr
/usr/share/hunspell/fr_CA
/usr/share/hunspell/fr_CH
/usr/share/hunspell/fr_MC
/usr/share/hunspell/fr_FR
/usr/share/hunspell/en_GB
/usr/share/hunspell/fr_BE
/usr/share/hunspell/en_US
/usr/share/myspell/dicts/hyph_en_CA
/usr/share/myspell/dicts/hyph_nl_NL
/usr/share/myspell/dicts/hyph_da_DK
/usr/share/myspell/dicts/en_ZA
/usr/share/myspell/dicts/fr_LU
/usr/share/myspell/dicts/hyph_es_ES
/usr/share/myspell/dicts/hyph_en_US
/usr/share/myspell/dicts/hyph_is_IS
/usr/share/myspell/dicts/fr_CA
/usr/share/myspell/dicts/hyph_fi_FI
/usr/share/myspell/dicts/hyph_ga_IE
/usr/share/myspell/dicts/hyph_pt_BR
/usr/share/myspell/dicts/hyph_sk_SK
/usr/share/myspell/dicts/hyph_uk_UA
/usr/share/myspell/dicts/fr_CH
/usr/share/myspell/dicts/fr_MC
/usr/share/myspell/dicts/hyph_el_GR
/usr/share/myspell/dicts/hyph_en_GB
/usr/share/myspell/dicts/en-GB
/usr/share/myspell/dicts/fr_FR
/usr/share/myspell/dicts/hyph_pt_PT
/usr/share/myspell/dicts/hyph_id_ID
/usr/share/myspell/dicts/en_GB
/usr/share/myspell/dicts/hyph_cs_CZ
/usr/share/myspell/dicts/fr_BE
LOADED DICTIONARY:
/usr/share/hunspell/en_US.aff
/usr/share/hunspell/en_US.dic
Hunspell 1.3.2
```
How can I tell `hunspell` to use fr_FR dictionnary so it is propagated to atom spell checker ?

**Comment by rugk** [comment]
That's weird. So on Windows there is no way to switch the language... :disappointed:
It would even be no problem if you could just change the path to the Hunspell files. As I already have them installed with Notepad++ spell checker plugin.

**Comment by KendallPark** [comment]
+1 Would love to use the medical dictionary installed on my computer. Need stuff like `gluconeogenesis` recognized.

**Comment by omninonsense** [comment]
+1 :+1: to fix this issue and add support for multiple languages.

**Comment by DimensionalScoop** [comment]
Also +1 for multi-language spell check!

**Comment by cganterh** [comment]
:+1:

**Comment by michaelsproul** [comment]
Please no more +1 comments, I subscribed to this issue so I'd receive updates but so far it's just been +1s.

**Comment by mnquintana** [comment]
Thanks for the feedback everyone! However, all these +1s and :+1:s are just adding noise, so to minimize that I'm going to lock this issue for now – we may unlock it once someone starts working on this enhancement to get further feedback.

**Comment by 50Wliu** [comment]
At the current pace of development, this feature will be available starting in Atom 1.11.0.
